modules/net-cloudnat/variables.tf

```diff
@@ -24,6 +24,22 @@ variable "config_min_ports_per_vm" {
   description = "Minimum number of ports allocated to a VM from this NAT config."
   type        = number
   default     = 64
+
+  validation {
+    condition = var.enable_dynamic_port_allocation
```

Comment:
I don't think you can check other variables from a validation condition, unless things changed recently... What about bundling both into a single variable of type object?